lib/cmdlib/instance_migration.py

```diff
@@ -953,7 +964,14 @@ def _ExecMigration(self):
         raise errors.OpExecError("Could not migrate instance %s: %s" %
                                  (self.instance.name, msg))
 
-      if result.payload.status != constants.HV_MIGRATION_ACTIVE:
+      if (postcopy_enabled
```

Comment:
Continuing from my previous comment, I feel we have moved decision-making to the wrong level. I understand that this is in part because we don't monitor migration progress at the backend level, we only poll it regularly from the master via the RPC and so it feels natural to have the master take the decision instead of the node.

That said, I feel your approach can be improved to avoid bringing low-level details such as the memory pass count in cmdlib. I see two ways out of this:
 - Move the whole low-level decision logic in KVM's `GetMigrationStatus` and signal that an instance should move to postcopy mode via `MigrationStatus` (either through a special status, or through another field instead of `dirty_sync_count`).
OR
 - Trigger the switchover to postcopy from inside `GetMigrationStatus` itself. This is a bit ugly because it introduces side-effects to a "Get" function, but it has a couple of advantages:
   - It saves some trivial decision-making on the master
   - It does not require a new RPC call
   - It closes the gap between the time we get the migration status and the time we issue the postcopy start